discovery/spec.md

```diff
@@ -924,52 +923,29 @@ Content-Type: application/json
 
 This MUST delete the Service at the referenced URL.
 
-The request MAY include a body.  Any such body MUST be an object containing an
-`id` matching the `{id}` in the path.  That object MAY include an `epoch`
-value.  If the Service corresponding with the specified `id` has a higher
-recorded `epoch` value than given, the request MUST fail.
-
-If other service attributes are included in the request, those MUST be
-ignored for the purposes of processing the request.
+The request URL MAY include an OPTIONAL `epoch` query parameter, and if
```

Comment:
This change reduces the consistency with bulk delete.  Not inherently a problem but the original form may be cleaner for clients.

Reply:
which consistency are you thinking of?

Reply:
Bulk deletes (i.e. on /services) have callers specifying `id`s in the body.  This change means this endpoint doesn't explicitly support the same convention.

Reply:
True but remember, this variant of the delete() in today's spec doesn't have a body either and the service in question is referenced in the URL - which means we're already using a different convention.

In general, while having a body on a delete() is REST legal, it is a bit unconventional IMO. And I thought that's one of the reasons we have this variant in the spec - ie. so people could do things as easily as `curl .../services/id` and nothing more. In that mode it's very natural to include more info in the URL. Having to add a body for just a single field adds quite a bit of complexity to an otherwise simple request.  And, TBH, I don't think people are going to be looking for consistency between these two. If they want consistency between a singleton delete() and a multi-service delete() then they're still free to use the other variant of delete() with a single service in the body.

Reply:
Adding the param seems like a win, especially for use cases involving curl or just a browser.  It's not clear that it is as natural to use one calling convention for one case and another for the other when writing code.

WRT consistency, I can accept that as a judgement call that is not clearly superior either way.